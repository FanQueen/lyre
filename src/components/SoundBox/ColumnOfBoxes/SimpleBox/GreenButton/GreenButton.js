import styled, { css } from 'styled-components';
import ButtonStyles from '../../../../../styles/ButtonStyles';

const GreenButton = styled.button`
  ${ButtonStyles} ${props => props.clicked && css`background: #98e1d6;`};
`;

export default GreenButton;
