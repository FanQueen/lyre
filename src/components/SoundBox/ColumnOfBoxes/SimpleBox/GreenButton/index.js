export { default } from './GreenButton.js';
