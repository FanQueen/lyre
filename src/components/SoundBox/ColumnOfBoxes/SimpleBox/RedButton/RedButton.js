import styled, { css } from 'styled-components';
import ButtonStyles from '../../../../../styles/ButtonStyles';

const RedButton = styled.button`
  ${ButtonStyles} ${props => props.clicked && css`background: #e2818a;`};
`;

export default RedButton;
