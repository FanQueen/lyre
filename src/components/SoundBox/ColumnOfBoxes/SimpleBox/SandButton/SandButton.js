import styled, { css } from 'styled-components';
import ButtonStyles from '../../../../../styles/ButtonStyles';

const SandButton = styled.button`
  ${ButtonStyles} ${props => props.clicked && css`background: #f0cc90;`};
`;

export default SandButton;
