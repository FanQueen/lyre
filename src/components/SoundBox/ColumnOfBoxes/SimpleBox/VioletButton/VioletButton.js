import styled, { css } from 'styled-components';
import ButtonStyles from '../../../../../styles/ButtonStyles';

const VioletButton = styled.button`
  ${ButtonStyles} ${props => props.clicked && css`background: #a8a2e8;`};
`;

export default VioletButton;
