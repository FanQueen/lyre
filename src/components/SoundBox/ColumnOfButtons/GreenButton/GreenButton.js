import { css } from 'styled-components';
import getButton from '../getButton';

const greenColor = css`#98e1d6`;
const GreenButton = getButton(greenColor);

export default GreenButton;
