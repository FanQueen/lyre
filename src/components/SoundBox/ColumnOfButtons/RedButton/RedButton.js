import { css } from 'styled-components';
import getButton from '../getButton';

const redColor = css`#e2818a`;
const RedButton = getButton(redColor);

export default RedButton;
