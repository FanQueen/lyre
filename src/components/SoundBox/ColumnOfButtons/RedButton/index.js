export { default } from './RedButton.js';
