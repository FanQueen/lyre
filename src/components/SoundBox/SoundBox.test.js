import React from 'react';
import { mount } from 'enzyme';
import SoundBox from './';
import RedButton from './ColumnOfButtons/RedButton';
import GreenButton from './ColumnOfButtons/GreenButton';
import SandButton from './ColumnOfButtons/SandButton';
import VioletButton from './ColumnOfButtons/VioletButton';

describe('SoundBox', () => {
  it('renders SoundBox with 4 colored columns', () => {
    const renderedSoundBox = mount(<SoundBox />);
    expect(renderedSoundBox.find(RedButton)).toHaveLength(4);
    expect(renderedSoundBox.find(GreenButton)).toHaveLength(4);
    expect(renderedSoundBox.find(SandButton)).toHaveLength(4);
    expect(renderedSoundBox.find(VioletButton)).toHaveLength(4);
  });
});
