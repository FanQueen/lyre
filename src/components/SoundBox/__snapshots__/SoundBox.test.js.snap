// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SoundBox renders SoundBox correctly 1`] = `
<SoundBox>
  <styled.div>
    <div
      className="sc-gzVnrw jHcFPD"
    >
      <withReducer(Component)
        audioFiles={
          Object {
            "arp": Array [
              "arp-1.wav",
              "arp-2.wav",
              "arp-3.wav",
              "arp-4.wav",
            ],
            "bass": Array [
              "bass-1.wav",
              "bass-2.wav",
              "bass-3.wav",
              "bass-4.wav",
            ],
            "beat": Array [
              "beat-1.wav",
              "beat-2.wav",
              "beat-3.wav",
              "beat-4.wav",
            ],
            "lead": Array [
              "lead-1.wav",
              "lead-2.wav",
              "lead-3.wav",
              "lead-4.wav",
            ],
          }
        }
        audioType="arp"
        clock={
           {
            "_clockNode": Object {
              "inputs": Array [],
              "name": "ScriptProcessorNode",
            },
            "_events": Array [],
            "_started": true,
            "context": Object {
              "inputs": Array [
                Object {
                  "inputs": Array [],
                  "name": "ScriptProcessorNode",
                },
              ],
              "name": "AudioDestinationNode",
            },
            "tickMethod": "ScriptProcessorNode",
            "toleranceEarly": 0.001,
            "toleranceLate": 0.1,
          }
        }
        colorButton={[Function]}
        key="0"
        soundBank={Object {}}
      >
        <Component
          audioFiles={
            Object {
              "arp": Array [
                "arp-1.wav",
                "arp-2.wav",
                "arp-3.wav",
                "arp-4.wav",
              ],
              "bass": Array [
                "bass-1.wav",
                "bass-2.wav",
                "bass-3.wav",
                "bass-4.wav",
              ],
              "beat": Array [
                "beat-1.wav",
                "beat-2.wav",
                "beat-3.wav",
                "beat-4.wav",
              ],
              "lead": Array [
                "lead-1.wav",
                "lead-2.wav",
                "lead-3.wav",
                "lead-4.wav",
              ],
            }
          }
          audioType="arp"
          clock={
             {
              "_clockNode": Object {
                "inputs": Array [],
                "name": "ScriptProcessorNode",
              },
              "_events": Array [],
              "_started": true,
              "context": Object {
                "inputs": Array [
                  Object {
                    "inputs": Array [],
                    "name": "ScriptProcessorNode",
                  },
                ],
                "name": "AudioDestinationNode",
              },
              "tickMethod": "ScriptProcessorNode",
              "toleranceEarly": 0.001,
              "toleranceLate": 0.1,
            }
          }
          colorButton={[Function]}
          dispatch={[Function]}
          soundBank={Object {}}
          state={
            Object {
              "selectedButton": null,
            }
          }
        >
          <styled.div>
            <div
              className="sc-bdVaJa cMfvsI"
            >
              <styled.button
                key="1"
                onClick={[Function]}
                selected={false}
              >
                <button
                  className="sc-htpNat kFDitE"
                  onClick={[Function]}
                  selected={false}
                />
              </styled.button>
              <styled.button
                key="2"
                onClick={[Function]}
                selected={false}
              >
                <button
                  className="sc-htpNat kFDitE"
                  onClick={[Function]}
                  selected={false}
                />
              </styled.button>
              <styled.button
                key="3"
                onClick={[Function]}
                selected={false}
              >
                <button
                  className="sc-htpNat kFDitE"
                  onClick={[Function]}
                  selected={false}
                />
              </styled.button>
              <styled.button
                key="4"
                onClick={[Function]}
                selected={false}
              >
                <button
                  className="sc-htpNat kFDitE"
                  onClick={[Function]}
                  selected={false}
                />
              </styled.button>
            </div>
          </styled.div>
        </Component>
      </withReducer(Component)>
      <withReducer(Component)
        audioFiles={
          Object {
            "arp": Array [
              "arp-1.wav",
              "arp-2.wav",
              "arp-3.wav",
              "arp-4.wav",
            ],
            "bass": Array [
              "bass-1.wav",
              "bass-2.wav",
              "bass-3.wav",
              "bass-4.wav",
            ],
            "beat": Array [
              "beat-1.wav",
              "beat-2.wav",
              "beat-3.wav",
              "beat-4.wav",
            ],
            "lead": Array [
              "lead-1.wav",
              "lead-2.wav",
              "lead-3.wav",
              "lead-4.wav",
            ],
          }
        }
        audioType="bass"
        clock={
           {
            "_clockNode": Object {
              "inputs": Array [],
              "name": "ScriptProcessorNode",
            },
            "_events": Array [],
            "_started": true,
            "context": Object {
              "inputs": Array [
                Object {
                  "inputs": Array [],
                  "name": "ScriptProcessorNode",
                },
              ],
              "name": "AudioDestinationNode",
            },
            "tickMethod": "ScriptProcessorNode",
            "toleranceEarly": 0.001,
            "toleranceLate": 0.1,
          }
        }
        colorButton={[Function]}
        key="1"
        soundBank={Object {}}
      >
        <Component
          audioFiles={
            Object {
              "arp": Array [
                "arp-1.wav",
                "arp-2.wav",
                "arp-3.wav",
                "arp-4.wav",
              ],
              "bass": Array [
                "bass-1.wav",
                "bass-2.wav",
                "bass-3.wav",
                "bass-4.wav",
              ],
              "beat": Array [
                "beat-1.wav",
                "beat-2.wav",
                "beat-3.wav",
                "beat-4.wav",
              ],
              "lead": Array [
                "lead-1.wav",
                "lead-2.wav",
                "lead-3.wav",
                "lead-4.wav",
              ],
            }
          }
          audioType="bass"
          clock={
             {
              "_clockNode": Object {
                "inputs": Array [],
                "name": "ScriptProcessorNode",
              },
              "_events": Array [],
              "_started": true,
              "context": Object {
                "inputs": Array [
                  Object {
                    "inputs": Array [],
                    "name": "ScriptProcessorNode",
                  },
                ],
                "name": "AudioDestinationNode",
              },
              "tickMethod": "ScriptProcessorNode",
              "toleranceEarly": 0.001,
              "toleranceLate": 0.1,
            }
          }
          colorButton={[Function]}
          dispatch={[Function]}
          soundBank={Object {}}
          state={
            Object {
              "selectedButton": null,
            }
          }
        >
          <styled.div>
            <div
              className="sc-bdVaJa cMfvsI"
            >
              <styled.button
                key="1"
                onClick={[Function]}
                selected={false}
              >
                <button
                  className="sc-ifAKCX hnEUDr"
                  onClick={[Function]}
                  selected={false}
                />
              </styled.button>
              <styled.button
                key="2"
                onClick={[Function]}
                selected={false}
              >
                <button
                  className="sc-ifAKCX hnEUDr"
                  onClick={[Function]}
                  selected={false}
                />
              </styled.button>
              <styled.button
                key="3"
                onClick={[Function]}
                selected={false}
              >
                <button
                  className="sc-ifAKCX hnEUDr"
                  onClick={[Function]}
                  selected={false}
                />
              </styled.button>
              <styled.button
                key="4"
                onClick={[Function]}
                selected={false}
              >
                <button
                  className="sc-ifAKCX hnEUDr"
                  onClick={[Function]}
                  selected={false}
                />
              </styled.button>
            </div>
          </styled.div>
        </Component>
      </withReducer(Component)>
      <withReducer(Component)
        audioFiles={
          Object {
            "arp": Array [
              "arp-1.wav",
              "arp-2.wav",
              "arp-3.wav",
              "arp-4.wav",
            ],
            "bass": Array [
              "bass-1.wav",
              "bass-2.wav",
              "bass-3.wav",
              "bass-4.wav",
            ],
            "beat": Array [
              "beat-1.wav",
              "beat-2.wav",
              "beat-3.wav",
              "beat-4.wav",
            ],
            "lead": Array [
              "lead-1.wav",
              "lead-2.wav",
              "lead-3.wav",
              "lead-4.wav",
            ],
          }
        }
        audioType="beat"
        clock={
           {
            "_clockNode": Object {
              "inputs": Array [],
              "name": "ScriptProcessorNode",
            },
            "_events": Array [],
            "_started": true,
            "context": Object {
              "inputs": Array [
                Object {
                  "inputs": Array [],
                  "name": "ScriptProcessorNode",
                },
              ],
              "name": "AudioDestinationNode",
            },
            "tickMethod": "ScriptProcessorNode",
            "toleranceEarly": 0.001,
            "toleranceLate": 0.1,
          }
        }
        colorButton={[Function]}
        key="2"
        soundBank={Object {}}
      >
        <Component
          audioFiles={
            Object {
              "arp": Array [
                "arp-1.wav",
                "arp-2.wav",
                "arp-3.wav",
                "arp-4.wav",
              ],
              "bass": Array [
                "bass-1.wav",
                "bass-2.wav",
                "bass-3.wav",
                "bass-4.wav",
              ],
              "beat": Array [
                "beat-1.wav",
                "beat-2.wav",
                "beat-3.wav",
                "beat-4.wav",
              ],
              "lead": Array [
                "lead-1.wav",
                "lead-2.wav",
                "lead-3.wav",
                "lead-4.wav",
              ],
            }
          }
          audioType="beat"
          clock={
             {
              "_clockNode": Object {
                "inputs": Array [],
                "name": "ScriptProcessorNode",
              },
              "_events": Array [],
              "_started": true,
              "context": Object {
                "inputs": Array [
                  Object {
                    "inputs": Array [],
                    "name": "ScriptProcessorNode",
                  },
                ],
                "name": "AudioDestinationNode",
              },
              "tickMethod": "ScriptProcessorNode",
              "toleranceEarly": 0.001,
              "toleranceLate": 0.1,
            }
          }
          colorButton={[Function]}
          dispatch={[Function]}
          soundBank={Object {}}
          state={
            Object {
              "selectedButton": null,
            }
          }
        >
          <styled.div>
            <div
              className="sc-bdVaJa cMfvsI"
            >
              <styled.button
                key="1"
                onClick={[Function]}
                selected={false}
              >
                <button
                  className="sc-bwzfXH kGMlao"
                  onClick={[Function]}
                  selected={false}
                />
              </styled.button>
              <styled.button
                key="2"
                onClick={[Function]}
                selected={false}
              >
                <button
                  className="sc-bwzfXH kGMlao"
                  onClick={[Function]}
                  selected={false}
                />
              </styled.button>
              <styled.button
                key="3"
                onClick={[Function]}
                selected={false}
              >
                <button
                  className="sc-bwzfXH kGMlao"
                  onClick={[Function]}
                  selected={false}
                />
              </styled.button>
              <styled.button
                key="4"
                onClick={[Function]}
                selected={false}
              >
                <button
                  className="sc-bwzfXH kGMlao"
                  onClick={[Function]}
                  selected={false}
                />
              </styled.button>
            </div>
          </styled.div>
        </Component>
      </withReducer(Component)>
      <withReducer(Component)
        audioFiles={
          Object {
            "arp": Array [
              "arp-1.wav",
              "arp-2.wav",
              "arp-3.wav",
              "arp-4.wav",
            ],
            "bass": Array [
              "bass-1.wav",
              "bass-2.wav",
              "bass-3.wav",
              "bass-4.wav",
            ],
            "beat": Array [
              "beat-1.wav",
              "beat-2.wav",
              "beat-3.wav",
              "beat-4.wav",
            ],
            "lead": Array [
              "lead-1.wav",
              "lead-2.wav",
              "lead-3.wav",
              "lead-4.wav",
            ],
          }
        }
        audioType="lead"
        clock={
           {
            "_clockNode": Object {
              "inputs": Array [],
              "name": "ScriptProcessorNode",
            },
            "_events": Array [],
            "_started": true,
            "context": Object {
              "inputs": Array [
                Object {
                  "inputs": Array [],
                  "name": "ScriptProcessorNode",
                },
              ],
              "name": "AudioDestinationNode",
            },
            "tickMethod": "ScriptProcessorNode",
            "toleranceEarly": 0.001,
            "toleranceLate": 0.1,
          }
        }
        colorButton={[Function]}
        key="3"
        soundBank={Object {}}
      >
        <Component
          audioFiles={
            Object {
              "arp": Array [
                "arp-1.wav",
                "arp-2.wav",
                "arp-3.wav",
                "arp-4.wav",
              ],
              "bass": Array [
                "bass-1.wav",
                "bass-2.wav",
                "bass-3.wav",
                "bass-4.wav",
              ],
              "beat": Array [
                "beat-1.wav",
                "beat-2.wav",
                "beat-3.wav",
                "beat-4.wav",
              ],
              "lead": Array [
                "lead-1.wav",
                "lead-2.wav",
                "lead-3.wav",
                "lead-4.wav",
              ],
            }
          }
          audioType="lead"
          clock={
             {
              "_clockNode": Object {
                "inputs": Array [],
                "name": "ScriptProcessorNode",
              },
              "_events": Array [],
              "_started": true,
              "context": Object {
                "inputs": Array [
                  Object {
                    "inputs": Array [],
                    "name": "ScriptProcessorNode",
                  },
                ],
                "name": "AudioDestinationNode",
              },
              "tickMethod": "ScriptProcessorNode",
              "toleranceEarly": 0.001,
              "toleranceLate": 0.1,
            }
          }
          colorButton={[Function]}
          dispatch={[Function]}
          soundBank={Object {}}
          state={
            Object {
              "selectedButton": null,
            }
          }
        >
          <styled.div>
            <div
              className="sc-bdVaJa cMfvsI"
            >
              <styled.button
                key="1"
                onClick={[Function]}
                selected={false}
              >
                <button
                  className="sc-bxivhb hTnTTK"
                  onClick={[Function]}
                  selected={false}
                />
              </styled.button>
              <styled.button
                key="2"
                onClick={[Function]}
                selected={false}
              >
                <button
                  className="sc-bxivhb hTnTTK"
                  onClick={[Function]}
                  selected={false}
                />
              </styled.button>
              <styled.button
                key="3"
                onClick={[Function]}
                selected={false}
              >
                <button
                  className="sc-bxivhb hTnTTK"
                  onClick={[Function]}
                  selected={false}
                />
              </styled.button>
              <styled.button
                key="4"
                onClick={[Function]}
                selected={false}
              >
                <button
                  className="sc-bxivhb hTnTTK"
                  onClick={[Function]}
                  selected={false}
                />
              </styled.button>
            </div>
          </styled.div>
        </Component>
      </withReducer(Component)>
    </div>
  </styled.div>
</SoundBox>
`;
